src/main/java/com/cst438/service/GradebookServiceProxy.java

```diff
@@ -0,0 +1,254 @@
+                section.setBuilding(dto.building());
+                section.setRoom(dto.room());
+                section.setTimes(dto.times());
+                section.setInstructor_email(dto.instructorEmail());
+                sectionRepository.save(section);
+                break;
+            case "deleteSection":
+                sectionRepository.deleteById(dto.secNo());
+                break;
+        }
+    }
+
+    private void handleUserAction(String action, String payload) {
+        UserDTO dto = fromJsonString(payload, UserDTO.class);
+        User user;
+        switch (action) {
+            case "addUser":
+                user = new User();
+                user.setId(dto.id());
+                user.setName(dto.name());
+                user.setEmail(dto.email());
+                user.setType(dto.type());
+                userRepository.save(user);
+                break;
+            case "updateUser":
+                user = userRepository.findById(dto.id()).orElseThrow(() ->
+                        new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
+                user.setName(dto.name());
+                user.setEmail(dto.email());
+                user.setType(dto.type());
+                userRepository.save(user);
+                break;
+            case "deleteUser":
+                userRepository.deleteById(dto.id());
+                break;
+        }
+    }
+
+    private void handleEnrollmentAction(String action, String payload) {
+        EnrollmentDTO dto = fromJsonString(payload, EnrollmentDTO.class);
+        Enrollment enrollment;
+        switch (action) {
+            case "addEnrollment":
+                enrollment = new Enrollment();
+                enrollment.setEnrollmentId(dto.enrollmentId());
+
+                Section section = sectionRepository.findById(dto.sectionNo()).orElseThrow(() ->
+                        new ResponseStatusException(HttpStatus.NOT_FOUND, "Section not found"));
+                enrollment.setSection(section);
+
+                User user = userRepository.findById(dto.studentId()).orElseThrow(() ->
+                        new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
+                enrollment.setUser(user);
+
+                enrollment.setGrade(dto.grade());
+                enrollmentRepository.save(enrollment);
+                break;
+            case "deleteEnrollment":
+                enrollmentRepository.deleteById(dto.enrollmentId());
+                break;
+        }
+    }
+
+    private void updateFinalGrade(GradeDTO dto) {
+        Enrollment enrollment = enrollmentRepository.findById(dto.gradeId()).orElseThrow(() ->
+                new ResponseStatusException(HttpStatus.NOT_FOUND, "Enrollment not found"));
+        enrollment.setGrade(String.valueOf(dto.score()));
+        enrollmentRepository.save(enrollment);
+    }
+
+    private void sendMessage(String action, Object obj) {
```

Comment:
This method should only take a string. Just use the method in the template he provided:
    private void sendMessage(String s) {
        rabbitTemplate.convertAndSend(gradebookServiceQueue.getName(), s);
    }